workers/unity/Assets/Playground/Scripts/UI/ConnectionScreenController.cs

```diff
@@ -99,7 +90,7 @@ public void OnConnectionFailed(string connectionError)
         {
             UnityObjectDestroyer.Destroy(worker);
             errorMessage = $"Connection failed. Please check the IP address entered.\nSpatialOS error message:\n{connectionError}";
-            isConnecting = false;
+            GUI.enabled = true;
```

Comment:
`GUI.enabled` should not change here. `GUI.enabled` should be reset as soon as possible within the OnGUI method. It's not really global state, it will affect other OnGUI methods with side effects otherwise. That was the purpose of the try/finally block.

See also https://docs.unity3d.com/ScriptReference/GUI-enabled.html for similar example. I guess you can just set it to true instead of storing and restoring.

It is possible that Unity now resets GUI.enabled before any `OnGUI` call (needs to be confirmed), if that is the case then it may not need to be reset, but either way it is incorrect to modify the field outside of `OnGUI` blocks e.g. in callbacks like this.

Reply:
OK it looks like it does get reset for each `OnGUI` so no need to reset it in a finally block, should be safe remove the `GUI.enabled` alterations anywhere that's not `OnGUI`.

Reply:
And because it's reset for every `OnGUI` call, `isConnecting` is still necessary so that it can be used to do, for every `OnGUI` call, `GUI.enabled = false`. 

In other words, if you set `GUI.enabled = false` in one frame and not the next (as it is done only in `TryConnect`) it will have no effect - because next `OnGUI` call it will be true, so the input fields and the buttons would still be interactive.